x-pack/plugins/spaces/public/spaces_manager/spaces_manager.ts

```diff
@@ -142,9 +144,19 @@ export class SpacesManager {
   public async getShareableReferences(
     objects: SavedObjectTarget[]
   ): Promise<SavedObjectsCollectMultiNamespaceReferencesResponse> {
-    return this.http.post(`/api/spaces/_get_shareable_references`, {
-      body: JSON.stringify({ objects }),
-    });
+    const response = await this.http.post<SavedObjectsCollectMultiNamespaceReferencesResponse>(
+      `/api/spaces/_get_shareable_references`,
+      { body: JSON.stringify({ objects }) }
+    );
+
+    // We should exclude any child-reference tags because we don't yet support reconciling/merging duplicate tags. In other words: tags can
+    // be shared directly, but if a tag is only included as a reference of a requested object, it should not be shared.
+    // To filter accordingly, we allow the first N objects of the response, where N is the number of requested objects, then filter out any
+    // tags after that.
+    const filteredObjects = response.objects.filter(
+      ({ type }, i) => i < objects.length || type !== TAG_TYPE
+    );
```

Comment:
This filtering approach feels fragile to me, since it relies on explicit object ordering in the response. I know we are relying on the ordering elsewhere, but would something like this be clearer?

```diff
diff --git a/x-pack/plugins/spaces/public/spaces_manager/spaces_manager.ts b/x-pack/plugins/spaces/public/spaces_manager/spaces_manager.ts
index baefa3d4cae..34e8a8bd896 100644
--- a/x-pack/plugins/spaces/public/spaces_manager/spaces_manager.ts
+++ b/x-pack/plugins/spaces/public/spaces_manager/spaces_manager.ts
@@ -151,10 +151,12 @@ export class SpacesManager {
 
     // We should exclude any child-reference tags because we don't yet support reconciling/merging duplicate tags. In other words: tags can
     // be shared directly, but if a tag is only included as a reference of a requested object, it should not be shared.
-    // To filter accordingly, we allow the first N objects of the response, where N is the number of requested objects, then filter out any
-    // tags after that.
+    const requestedObjectsSet = objects.reduce(
+      (acc, object) => acc.add(`${object.type}:${object.id}`),
+      new Set<string>()
+    );
     const filteredObjects = response.objects.filter(
-      ({ type }, i) => i < objects.length || type !== TAG_TYPE
+      ({ type, id }, i) => type !== TAG_TYPE || requestedObjectsSet.has(`tag:${id}`)
     );
     return { objects: filteredObjects };
   }

```